### PR TITLE
CP-16626 Preparation for using a different unikernel VM

### PR DESCRIPTION
**Pull request by lindig**
* Improved documentation how to set up Testarossa
* Some code cleanup; use `string` rather than `bytes`; better use of pattern matching
* Preparation to use a different unikernel VM for testing (but don't do it yet)
* Script to download binary xen-test.vm to be used

Some commit messages use a wrong prefix in the ticket number: CA-16626 should have been CP-16626.

### CODE REVIEW

**Comment by lindig** [comment]
I have now integrated my branch `native` that uses compilation to native code and documented `test`common` in an interface. I see no advantage using OCamlScript given the very fast build times and the added benefit of having types available in editors thanks to annotation file produced by the compiler.

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#
```
Comment (euanh) [inline Makefile]: Plesae use oasis instead of introducing yet another Makefile variant into our code.

Comment (lindig) [inline Makefile]: I have moved to Oasis in my `refactor-common` branch.

**Comment by jonludlam** [comment]
Closing as discussed - this will be superseded.

**Pull request closed without merging**